pkg/audit/cursor.go

```diff
@@ -128,11 +110,43 @@ func getRandomStripe(pointer *pb.Pointer) (index int64, err error) {
 	return randomStripeIndex.Int64(), nil
 }
 
-func getRandomPointer(pointerItems []*pb.ListResponse_Item) (pointer *pb.ListResponse_Item, err error) {
+// getRandomPointer attempts to get a random pointer from a list, and if the selected pointer is invalid, it returns a list without that item
+func (cursor *Cursor) getRandomPointer(pointerItems []*pb.ListResponse_Item) (pointer *pb.Pointer, newPointerItems []*pb.ListResponse_Item, err error) {
```

Comment:
Maybe rename to reflect that its getting a particular type of "random" now

Reply:
possibly also break this logic into smaller pieces... delete expired items maybe be conceptually correct but when its just slipped in there its like "what the heck!?"

Reply:
renamed to `getRandomValidPointer`

I am still thinking of an ideal way to remove the delete logic. It is hard to know how to separate it without making things more complex